src/PureCloud.Client/Apis/TelephonyProvidersEdgeApi.cs

```diff
@@ -5,64 +5,59 @@
 using PureCloud.Client.Http;
 using PureCloud.Client.Json;
 using PureCloud.Client.Models;
+using PureCloud.Client.Extensions;
 
 namespace PureCloud.Client.Apis;
 
 /// <inheritdoc />
 public sealed class TelephonyProvidersEdgeApi : ITelephonyProvidersEdgeApi
 {
-    private readonly HttpClient _httpClient;
+    private readonly IHttpClientFactory _httpClientFactory;
     private readonly PureCloudJsonSerializerOptions _options;
 
     public TelephonyProvidersEdgeApi(IHttpClientFactory httpClientFactory, IOptions<PureCloudJsonSerializerOptions> options)
     {
-        _httpClient = httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+        _httpClientFactory = httpClientFactory;
         _options = options.Value;
     }
 
     /// <inheritdoc />
 
     public async Task<Edge> GetEdgeAsync(string edgeId, CancellationToken cancellationToken = default)
     {
-        ArgumentException.ThrowIfNullOrEmpty(nameof(edgeId), edgeId);
+        ArgumentException.ThrowIfNullOrEmpty(edgeId);
 
-        var uri = UriHelper.GetUri($"/api/v2/telephony/providers/edges/{edgeId}", null);
-        var response = await _httpClient.GetAsync(uri, cancellationToken);
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var response = await client.GetAsync($"/api/v2/telephony/providers/edges/{Uri.EscapeDataString(edgeId)}", cancellationToken);
         response.EnsureSuccessStatusCode();
 
         return await response.Content.ReadFromJsonAsync<Edge>(_options.JsonSerializerOptions, cancellationToken);
     }
 
     /// <inheritdoc />
 
-    public async Task DeleteEdgeAsync(string edgeId, CancellationToken cancellationToken = default)
+    public async Task<bool> DeleteEdgeAsync(string edgeId, CancellationToken cancellationToken = default)
     {
-        ArgumentException.ThrowIfNullOrEmpty(nameof(edgeId), edgeId);
-
-        var uri = UriHelper.GetUri($"/api/v2/telephony/providers/edges/{edgeId}", null);
-        var response = await _httpClient.DeleteAsync(uri, cancellationToken);
-        response.EnsureSuccessStatusCode();
-    }
+        ArgumentException.ThrowIfNullOrEmpty(edgeId);
 
-    /// <inheritdoc />
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
 
-    public async Task<EdgeEntityListing> GetEdgesAsync(CancellationToken cancellationToken = default)
-    {
-        var uri = UriHelper.GetUri("/api/v2/telephony/providers/edges", null);
-        var response = await _httpClient.GetAsync(uri, cancellationToken);
+        var response = await client.DeleteAsync($"/api/v2/telephony/providers/edges/{Uri.EscapeDataString(edgeId)}", cancellationToken);
         response.EnsureSuccessStatusCode();
 
-        return await response.Content.ReadFromJsonAsync<EdgeEntityListing>(_options.JsonSerializerOptions, cancellationToken);
+        return true;
```

Comment:
Retun IsSuccessStausCode instead of EnsureSuccessStatusCode(). Apply this everywhere we return bool